libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -674,7 +674,7 @@ protected void onPostExecute(OAuth2.TokenEndpointResponse tr) {
             final SalesforceSDKManager mgr = SalesforceSDKManager.getInstance();
 
             // Failure cases.
-            if (isSalesforceIntegrationUser) {
+            if (mgr.shouldBlockSalesforceIntegrationUser() && isSalesforceIntegrationUser) {
```

Comment:
is it possible to check `mgr.shouldBlockSalesforceIntegrationUser()` before making the extra API call to get if the user is an integration user?